modules/blockfrost.nix

```
@@ -0,0 +1,82 @@
+{
+  config,
+  lib,
+  ...
+}: let
+  cfg = config.cardano.blockfrost;
+  dbsync-cfg = config.services.cardano-db-sync;
+  inherit (lib) mkIf mkMerge mkEnableOption;
+in {
+  options.cardano.blockfrost =
+    {
+      enable =
+        mkEnableOption ''
+        '';
+    }
+    // {default = config.cardano.enable or false;};
+
+  config = mkIf cfg.enable (mkMerge [
+    {
+      services.blockfrost = {
+        enable = true;
+        settings = {
+          inherit (config.cardano) network;
+          dbSync = {
+            inherit (dbsync-cfg.postgres) user port database;
```

```suggestion
            user = mkIf (dbsync-cfg.enabled or false) dbsync-cfg.postgres.user; 
            ...
```
we have to check if the module exists and is enabled